### PR TITLE
Remove @next and prefer @react as the default

### PR DESCRIPTION
**Pull request by ericclemmons**
*Issue #, if available:* https://app.asana.com/0/0/1200974176066565/f

*Description of changes:*

This PR **removes usage of Next.js as a "framework" and instead only refers to React, Angular, or Vue**.  (So not supersets like Nuxt.js, Ionic, etc.)

- [x] Replace `@next` tags with `@react`
- [x] Remove unused `.next.mdx` pages

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/e2e/features/ui/components/authenticator/verify-user.feature [hunk 1]
```diff
@@ -7,14 +7,15 @@ Feature: Verify User
   Background:
     Given I'm running the example "ui/components/authenticator/sign-in-with-email"
 
-  @angular @next @vue
```
Comment (wlee221) [inline packages/e2e/features/ui/components/authenticator/verify-user.feature:10]: @slaymance just merged #369 that fixes account verification -- can you try adding these again?

**Review by dbanksdesign — APPROVED** [review]
LGTM! :shipit:

File: packages/e2e/features/ui/components/authenticator/verify-user.feature (hunk 1 above)
Comment (ericclemmons) [reply packages/e2e/features/ui/components/authenticator/verify-user.feature:10]: I resolved the conflicts, but perhaps incorrectly with @6b4ebc.

File: packages/e2e/features/ui/components/authenticator/verify-user.feature [hunk 2]
```diff
@@ -7,14 +7,15 @@ Feature: Verify User
...
+  # Verify not functioning in Angular
+  @react @vue
```
Comment (ericclemmons) [inline packages/e2e/features/ui/components/authenticator/verify-user.feature:11]: @wlee221 Based on #369 this should've been

```suggestion
  @angular @react @vue
```

File: README.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ Amplify UI 2.0 is an open-source UI library with cloud-connected components that
 
 ## Getting started
 
-We will be publishing the next major version (2.0) of these packages under the `@next` tag in npm when they are ready:
+We will be publishing the next major version (2.0) of these packages under the `@react` tag in npm when they are ready:
```
Comment (reesscot) [inline README.md:29]: @ericclemmons Was this change intentional? Won't we still publish to the next github tag?

File: packages/e2e/features/ui/components/authenticator/verify-user.feature (hunk 2 above)
Comment (wlee221) [reply packages/e2e/features/ui/components/authenticator/verify-user.feature:11]: Let's go for it!

File: packages/e2e/features/ui/components/authenticator/verify-user.feature [hunk 4]
```diff
@@ -7,14 +7,15 @@ Feature: Verify User
...
   Scenario: Redirect to "Verify" page
     When I type my "email" with status "UNVERIFIED"
     And I type my password
     And I click the "Sign in" button
     Then I will be redirected to the verify user page
 
-  @angular @next @vue
+  @react @vue
```
Comment (wlee221) [inline packages/e2e/features/ui/components/authenticator/verify-user.feature:18]:
```suggestion
  @angular @react @vue
```

File: README.md (hunk 3 above)
Comment (ericclemmons) [reply README.md:29]: Find/replace mistake. Thanks! 

```suggestion
We will be publishing the next major version (2.0) of these packages under the `@next` tag in npm when they are ready:
```

File: packages/e2e/features/ui/components/authenticator/verify-user.feature (hunk 1 above)
Comment (ericclemmons) [reply packages/e2e/features/ui/components/authenticator/verify-user.feature:10]: Looks like you fixed it! Thanks!

**Review by reesscot — APPROVED** [review]

**Review by wlee221 — COMMENTED** [review]
It looks like next tests are passing vacuously because they are expecting @next tags: https://github.com/aws-amplify/amplify-ui/pull/371/checks?check_run_id=3602460832.

https://github.com/aws-amplify/amplify-ui/blob/c30f17022ad35c034d82a8695825cbe30acc6ae1/.github/workflows/tests.yml#L58 should be `- 'react'` if I'm not mistaken.

**Comment by ericclemmons** [comment]
@wlee221 Good catch – I was afraid of that.

Given the option of renaming `next-example` to `react-example`, I decided to instead let our e2e test indicate which tags are relevant to it. (`next` framework supports `@react` tests)

**Review by wlee221 — APPROVED** [review]
🚢 LGTM!

**Comment by ericclemmons** [comment]
All tests passed, but some old runs were stuck in limbo:
> Screen Shot 2021-09-14 at 2 52 53 PM

**Comment by wlee221** [comment]
@ericclemmons Yup, I just removed "required" tag off those tags.
